tools/image-tool/src/docker.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::{Path, PathBuf};
+
+use async_trait::async_trait;
+
+use crate::{ImageToolImpl, Result};
+
+pub struct DockerCLI {
+    pub(crate) path: PathBuf,
+}
+
+#[async_trait]
+impl ImageToolImpl for DockerCLI {
+    fn path(&self) -> PathBuf {
+        self.path.clone()
+    }
+
+    async fn pull_archive(&self, file: &Path, uri: &str, _cache: &Path) -> Result<()> {
+        // First we pull the image to local daemon
+        self.spawn(
+            &["pull", uri],
+            format!("failed to pull image to local docker from {}", uri),
+        )
+        .await?;
+        // Now we can use docker save to save the archive
+        let archive_path = file.to_string_lossy();
+        self.spawn(
+            &["save", uri, "-o", archive_path.as_ref()],
+            format!(
+                "failed to save image archive from {} to {}",
+                uri,
+                file.display()
+            ),
+        )
+        .await?;
+        Ok(())
+    }
+
+    async fn get_manifest(&self, uri: &str) -> Result<Vec<u8>> {
+        self.output(
+            &["manifest", "inspect", uri],
+            format!("failed to inspect manifest of resource at {}", uri),
+        )
+        .await
+    }
+
+    async fn get_config(&self, uri: &str) -> Result<Vec<u8>> {
```

Comment:
Per https://github.com/bottlerocket-os/twoliter/pull/306 I suppose we have to pull first :(